### PR TITLE
docs: Remove mentoring related docs and issue template

### LINKED ISSUE #11802: Remove mentoring docs/issue template
I am considering removing the mentoring option and related docs going forward. Any objections?

Reasons:
* We got a lot of mentoring requests and do not have many active mentors.
* They often become stale and frustrate new contributors.
* We should encourage people to comment on existing issues that they are interested in working on.

### PR DESCRIPTION
**Pull request by terrytangyuan**
Closes #11802

### CODE REVIEW

File: docs/mentoring.md [hunk 1]
```diff
@@ -1,60 +0,0 @@
-# Mentoring
```
Comment (agilgur5) [inline docs/mentoring.md:1]: do we want to rewrite this page to instead encourage questions on issues and PRs? I can rewrite it if so

File: .github/ISSUE_TEMPLATE/mentoring_request.md [hunk 2]
```diff
@@ -1,22 +0,0 @@
----
```
Comment (agilgur5) [inline .github/ISSUE_TEMPLATE/mentoring_request.md:1]: we could potentially change this to a redirect (via the `config.yml`) to the significantly altered mentoring docs that I mentioned above

**Review by agilgur5 — COMMENTED** [review]
do we want to remove these in entirety? or instead narrow the scope significantly. or encourage users to ask questions on issues and PRs? (I am in favor of the latter)

**Comment by terrytangyuan** [comment]
I'd like to remove these entirely as mentioned in https://github.com/argoproj/argo-workflows/issues/11802

**Comment by juliev0** [comment]
thank you for doing this

**Review by juliev0 — APPROVED** [review]

**Comment by agilgur5** [comment]
Updating here that I've gone through all open & closed mentoring requests and properly labeled them, closed them if they were inactive, and left updates.

There are 5 leftover that are still open as they all had some work on them or were in progress.

**Pull request merged**